Simplified/NYPLBookCoverRegistry.m

```diff
@@ -85,6 +85,29 @@ - (NSURL *)URLForPinnedThumbnailImageOfBookIdentifier:(NSString *const)bookIdent
           URLByAppendingPathComponent:[bookIdentifier SHA256]];
 }
 
+- (void)coverImageForBook:(NYPLBook *)book handler:(void (^)(UIImage *image))handler
+{
+
+  //Thumbnail first as placeholder
+  [self thumbnailImageForBook:book handler:handler];
```

Comment:
I'm not sure what you mean by "side effects". Right when the detail view appears, it begins to download the larger cover photo, but the first handler returns the cached thumbnail, so that there's a nice transition between the low-res and higher-res images. You can visually see this with a 1-2 second delay even on fast connections, so both handlers are definitely being used.. 

I'm definitely open to a better way of approaching it however.. I just figured to piggyback on the local cache / generated cover logic of the original method rather than to flesh it out fully in two different places.